codegen/src/test/resources/software/amazon/awssdk/codegen/poet/rules/endpoint-resolve-interceptor.java

```diff
@@ -125,9 +125,6 @@ private static void operationWithStaticContextParamsStaticContextParams(QueryEnd
     private <T extends Identity> SelectedAuthScheme<T> authSchemeWithEndpointSignerProperties(
         List<EndpointAuthScheme> endpointAuthSchemes, SelectedAuthScheme<T> selectedAuthScheme) {
         for (EndpointAuthScheme endpointAuthScheme : endpointAuthSchemes) {
-            if (!endpointAuthScheme.schemeId().equals(selectedAuthScheme.authSchemeOption().schemeId())) {
```

Comment:
Thanks @millems for highlighting that not checking this could be an issue for useSraAuth=true client, in a case for example where SelectedAuthScheme=sigv4a, but endpointAuthSchemes=["sigv4","sigv4a"] and the shared signer properties like disableDoubleEncoding have different values from endpoint rules. Without this check, the value from sigv4 EndpointAuthScheme will get used, even though it should use from sigv4a EndpointAuthScheme.

Will address this is separate PR.

Reply:
addressing in https://github.com/aws/aws-sdk-java-v2/pull/4505